pkg/model/participation/status.go

```diff
@@ -20,31 +20,51 @@ type QuestionStatus struct {
 	Answers []*AnswerStatus `json:"answers"`
 }
 
+// StakingStatus holds the status of a staking.
+type StakingStatus struct {
+	Staked   uint64 `json:"staked"`
+	Rewarded uint64 `json:"rewarded"`
+	Symbol   string `json:"symbol"`
+}
+
 // EventStatus holds the status of the event
 type EventStatus struct {
 	MilestoneIndex milestone.Index   `json:"milestoneIndex"`
 	Status         string            `json:"status"`
 	Questions      []*QuestionStatus `json:"questions,omitempty"`
+	Staking        *StakingStatus    `json:"staking,omitempty"`
 	Checksum       string            `json:"checksum"`
 }
 
 // EventStatus returns the EventStatus for an event with the given eventID.
-func (pm *ParticipationManager) EventStatus(eventID EventID) (*EventStatus, error) {
-
-	confirmedMilestoneIndex := pm.syncManager.ConfirmedMilestoneIndex()
-
+func (pm *ParticipationManager) EventStatus(eventID EventID, milestone ...milestone.Index) (*EventStatus, error) {
 	event := pm.Event(eventID)
 	if event == nil {
 		return nil, ErrEventNotFound
 	}
 
+	index := pm.syncManager.ConfirmedMilestoneIndex()
+	if index > event.EndMilestoneIndex() {
+		index = event.EndMilestoneIndex()
+	}
+
+	if len(milestone) > 0 {
+		index = milestone[0]
+	}
```

Comment:
Maybe do this check after you applied the passed parameter?
```suggestion
	index := pm.syncManager.ConfirmedMilestoneIndex()
	if len(milestone) > 0 {
		index = milestone[0]
	}

	if index > event.EndMilestoneIndex() {
		index = event.EndMilestoneIndex()
	}
```